### PR TITLE
Test against Python 3.12dev

### PR DESCRIPTION
**Pull request by larrybradley**

### CODE REVIEW

**Comment by larrybradley** [comment]
The deprecation warning is coming from matplotlib pulling in `dateutil` (https://github.com/spacetelescope/pypolyclip/actions/runs/6242732805/job/16947115105?pr=13#step:5:111).  Nothing I can do, so ignoring that warning for now.   CC: @pllim

File: pyproject.toml [hunk 1]
```diff
@@ -62,6 +62,10 @@ testpaths = [
 addopts = '--color=yes'
 xfail_strict = true
 remote_data_strict = true
+filterwarnings = [
+    'error',
+    'ignore:datetime.datetime.utcfromtimestamp() is deprecated:DeprecationWarning',
```
Comment (pllim) [inline pyproject.toml:67]: Escape for regex?
```suggestion
    'ignore:datetime.datetime.utcfromtimestamp\(\) is deprecated:DeprecationWarning',
```

Comment (larrybradley) [reply pyproject.toml]: yep, thanks

File: pyproject.toml [hunk 2]
```diff
@@ -62,6 +62,10 @@ testpaths = [
 addopts = '--color=yes'
 xfail_strict = true
 remote_data_strict = true
+filterwarnings = [
+    'error',
+    'ignore:datetime.datetime.utcfromtimestamp\(\) is deprecated:DeprecationWarning',
```
Comment (pllim) [inline pyproject.toml:67]: If you want to be pedantic but CI looks already green.
```suggestion
    'ignore:datetime\.datetime\.utcfromtimestamp\(\) is deprecated:DeprecationWarning',
```

**Review by pllim — APPROVED** [review]
You might also want to report upstream to matplotlib?

File: pyproject.toml (hunk 2 above)
Comment (larrybradley) [reply pyproject.toml:67]: I suppose we wouldn't catch if another character replaces the ".", but that's not likely to happen.  In any case, this line is only temporary until it's fixed upstream, so I'll go ahead and merge as is.

**Comment by pllim** [comment]
BTW, looks like it is patched upstream at https://github.com/dateutil/dateutil/pull/1285 , so I guess we'll just have to wait it out.

**Pull request merged**